Shared/Samples/Create KML multi-track/CreateKMLMultiTrackView.swift

```diff
@@ -0,0 +1,185 @@
+                
+                do {
+                    switch asyncAction {
+                    case .recordTrack:
+                        for await location in model.locationDisplay.$location where location != nil {
+                            model.addTrackElement(atPoint: location!.position)
+                            statusText = "Recording KML track. Elements added: \(model.trackElements.count)"
+                        }
+                        
+                        statusText = "Tap record to capture KML track elements."
+                    case .saveKMLMultiTrack:
+                        await model.locationDisplay.dataSource.stop()
+                        
+                        try await model.saveKMLMultiTrack()
+                        multiTrack = try await model.loadKMLMultiTrack()
+                        
+                        statusText = "Saved KML multi-track to 'HikingTracks.kmz'."
+                    case .reset:
+                        model.reset()
+                        multiTrack = nil
+                        await mapViewProxy.setViewpointScale(model.locationDisplay.initialZoomScale)
+                        
+                        fallthrough
+                    case .startNavigation:
+                        try await model.startNavigation()
+                        statusText = "Tap record to capture KML track elements."
+                    }
+                } catch {
+                    self.error = error
+                }
+            }
+            .task {
+                // Monitors the auto pan mode to determine if recenter button should be enabled.
+                for await autoPanMode in model.locationDisplay.$autoPanMode {
+                    recenterIsEnabled = autoPanMode != .navigation
+                }
+            }
+            .errorAlert(presentingError: $error)
+        }
+    }
+}
+
+private extension CreateKMLMultiTrackView {
+    /// A picker for selecting a track from a KML multi-track.
+    struct TrackPicker: View {
+        /// The KML tracks options shown in the picker.
+        private let tracks: [KMLTrack]
+        
+        /// The proxy for setting map view's viewpoint to the selected track's geometry.
+        private let mapViewProxy: MapViewProxy
+        
+        /// The track selected by the picker.
+        @State private var selectedTrack: KMLTrack?
+        
+        init(multiTrack: KMLMultiTrack, mapViewProxy: MapViewProxy) {
+            tracks = multiTrack.tracks
+            self.mapViewProxy = mapViewProxy
+        }
+        
+        var body: some View {
+            Picker("Track", selection: $selectedTrack) {
+                Text("All Tracks")
+                    .tag(nil as KMLTrack?)
+                
+                ForEach(Array(tracks.enumerated()), id: \.offset) { offset, track in
+                    Text("KML Track #\(offset + 1)")
+                        .tag(track)
+                }
```

Comment:
This is fine for a list like this that doesn't change, but you wouldn't want to use indices/offsets for a mutable list.